File: packages/mobile_flutter/lib/utils/validation.dart
import '../config/app_config.dart';

bool isValidImageSize(int fileSize) {
  return fileSize <= AppConfig.imageMaxSizeMb * 1024 * 1024;
}

String getAccuracyLevel(double accuracy) {
  if (accuracy <= AppConfig.gpsAccuracyGood) return 'good';
  if (accuracy <= AppConfig.gpsAccuracyModerate) return 'moderate';
  return 'poor';
}

bool isValidEmployeeId(String id) {
  return id.isNotEmpty && id.length <= 20;
}

bool isValidPassword(String password) {
  return password.length >= 8;
}

File: packages/mobile/src/utils/validation.ts
import { Config } from '../constants/config';

export function isValidImageSize(fileSize: number): boolean {
  return fileSize <= Config.IMAGE_MAX_SIZE_MB * 1024 * 1024;
}

export function getAccuracyLevel(
  accuracy: number,
): 'good' | 'moderate' | 'poor' {
  if (accuracy <= Config.GPS_ACCURACY_GOOD) return 'good';
  if (accuracy <= Config.GPS_ACCURACY_MODERATE) return 'moderate';
  return 'poor';
}

export function isValidEmployeeId(id: string): boolean {
  return id.length > 0 && id.length <= 20;
}

export function isValidPassword(password: string): boolean {
  return password.length >= 8;
}
